biz.aQute.remote/src/aQute/remote/util/AgentSupervisor.java

```diff
@@ -41,16 +42,28 @@
 	}
 
 	protected void connect(Class<Agent> agent, Supervisor supervisor, String host, int port) throws Exception {
-		while (true)
+		connect(agent, supervisor, host, port, -1);
+	}
+
+	protected void connect(Class<Agent> agent, Supervisor supervisor, String host, int port, int timeout)
```

Comment:
Any negative number is the same as -1. It is an indefinite wait. The alternative is to check timeout value on entrance to the method and throw an exception if timeout < -1.

Looking at `Object.wait(long)` it uses 0 for indefinite wait. Negative value is an error.